src/Components/TaskManagerComponent/TaskManagerComponent/index.ts

```diff
@@ -1,142 +1,145 @@
-	 */
-	public getOutputs(): IOutputs {
-		return {};
-	}
-
-	/**
-	 * Called when the control is to be removed from the DOM tree. Controls should use this call for cleanup.
-	 * i.e. cancelling any pending remote calls, removing listeners, etc.
-	 */
-	public destroy(): void {
-		// Add code to cleanup control if necessary
-	}
-}
\ No newline at end of file
+    public container: HTMLDivElement;
+    private _props: ITaskManagerProps;
+
+    /**
+     * Empty constructor.
+     */
+    constructor() {}
+
+    /**
+     * Used to initialize the control instance. Controls can kick off remote server calls and other initialization actions here.
+     * Data-set values are not initialized here, use updateView.
+     *
+     * @param context The entire property bag available to control via Context Object; It contains values as set up by the customizer mapped to property names defined in the manifest, as well as utility functions.
+     * @param notifyOutputChanged A callback method to alert the framework that the control has new outputs ready to be retrieved asynchronously.
+     * @param state A piece of data that persists in one session for a single user. Can be set at any point in a controls life cycle by calling 'setControlState' in the Mode interface.
+     * @param container If a control is marked control-type='standard', it will receive an empty div element within which it can render its content.
+     */
+    public init(
+        context: ComponentFramework.Context<IInputs>,
+        notifyOutputChanged: () => void,
+        state: ComponentFramework.Dictionary,
+        container: HTMLDivElement
+    ): void {
+        // Add control initialization code
+        this.container = container;
+    }
+
+    /**
+     * Called when any value in the property bag has changed. This includes field values, data-sets, global values such as container height and width, offline status, control metadata values such as label, visible, etc.
+     *
+     * @param context The entire property bag available to control via Context Object; It contains values as set up by the customizer mapped to names defined in the manifest, as well as utility functions
+     */
+    public async updateView(context: ComponentFramework.Context<IInputs>): Promise<void> {
+        this._props = {
+            tasks: this.mapTasks(context.parameters.tasks),
+            panelTitle: Utils.extractMultilingualText(context.parameters.panelTitle.raw || "", context.userSettings.languageId),
+            context: context,
+            badgeConfig: await this.loadBadgeConfiguration(context)
+        };
+
+        ReactDOM.render(React.createElement(TaskManager, this._props), this.container);
+    }
+
+    /**
+     * It is called by the framework prior to a control receiving new data.
+     *
+     * @returns an object based on nomenclature defined in manifest, expecting object[s] for property marked as “bound” or “output”
+     */
+    public getOutputs(): IOutputs {
```

Comment:
are these two functions necessary if they don't do anything?

Reply:
It's required for the PCF.